### ISSUE TITLE
windows() function open low resolution graphic window on high DPI display.

### ISSUE THREAD

**Issue by Marchen** [Microsoft/RTVS#2018]
From RTVS, windows() opens a blurred device like

![rtvs](https://cloud.githubusercontent.com/assets/3863577/16400122/94e9ed92-3d13-11e6-85f8-98ae1e6fa3d3.PNG)

From MRO, 

![mro](https://cloud.githubusercontent.com/assets/3863577/16400143/c09b70b4-3d13-11e6-9ffb-614110aa1612.png)

Note "Disable display scaling on high DPI settings" of compatibility setting of MRO is on.

**Comment by huguesv** [comment]
Hi Marchen,

I tried it out, and I was able to disable display scaling when running in RTVS by changing the compatibility setting for:
"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\R Tools for Visual Studio\Microsoft.R.Host.exe"

PS: Sorry for the delay, I've just returned from a vacation.

**Comment by Marchen** [comment]
Hi Hugues,

Thanks for your help. I could set compatibility setting and could use high DPI plot window.
Is it possible to make external plot window (i.e., window made by `windows()` function) to use Windows DPI setting like internal plot window?

**Comment by huguesv** [comment]
For the internal plot window, we use `png()` where we pass in a `res` that we adjust based on DPI setting (96 for 100%, 144 for 150%, etc).
There is no `res` parameter for `windows()`, but you can set a larger font size with `pointsize` parameter, and that gives similar result.

**Comment by Marchen** [comment]
Thanks for the comment and I'm sorry for my late reply.

If I use `pointsize` parameter of `windows()`, I can set a larger font size, but the resolution of the plot window is still low.
I think it is better to control resolution of the plot window by modifying the compatibility option of the R Host.
Is it difficult for RTVS to automatically set the compatibility setting of Microsoft.R.Host.exe (or use manifest file or something) to adjust DPI setting of plot window?

**Comment by huguesv** [comment]
I think it would be fine if we added the necessary manifest for Microsoft.R.Host.exe.  I was worried that the Save Variable and Get From Variable UI wouldn't respect the DPI settings, but it looks like they do.

I'm curious -- what makes you want to use the `windows()` plots instead of the RTVS plot window?

What could we add or fix in RTVS plots (other than that bug you filed) that would make the RGui window unnecessary in your work flow?

**Comment by Marchen** [comment]
Hi Hugues,

Thanks for your comment.
I'm greatly loving functionality of the RTVS plot window, but in my work flow, sometimes I need external windows.

First, sometimes I want to see several graphs together and think about meaning of the graphs. 
For such purpose making multiple graphs by `windows()` function and aligning them like usual window and switching with task bar is more comfortable for me.
I know RTVS plot window provides graph history function, but for this purpose, multiple windows by `windows()` function are more useful for me.

Second, sometimes I put several graphs into a single plot window (e.g., using `par(mfrow = c(5, 5))`).
Although usually I'm using plot window pinned with R editor, I need to enlarge plot window to see such a complicated graph because I'm using a laptop with 13'' display.
I found I can maximize the RTVS plot window by double clicking the title bar.
But if I maximize the window, it completely hides the editor window.
So switching editor window and plot window again and again is more difficult (or seems to require more steps) than using external window.

Finally, when I make a graph for some publication, its shape (i.e., height and width of the window) is not similar to that of pinned window.
For this case, switching editor window and external plot window again and again is important because I need to edit graphic parameter again and again to make good looking graph.
Therefore, external window that is easily hidden under the editor window is more preferable for this case.

In conclusion, if RTVS provides a way to create multiple plot window and provide a way to hide large plot window under the editor, maybe I don't need external window made by `windows()`.


Michio

**Comment by huguesv** [comment]
Thank you very much for taking the time to write this down.
Multiple plot windows is already very high on our TODO list.
As for hiding a large plot window in the background, we never realized that this was a problem, so it's great to have that feedback :)

**Comment by huguesv** [comment]
Fixed for RTVS 0.5: `windows()` devices should no longer be blurry when used from RTVS.
